### ISSUE TITLE
Don't work when .git placed in parent directory

### ISSUE THREAD

**Issue by galkinrost** [observing/pre-commit#27]
Hi! Thanks for amazing solution of hooks implementation for node.js. But could you add support for cases when .git placed in parent directory relative to package.json?

**Comment by tuhoojabotti** [comment]
:+1:

**Comment by davidchase** [comment]
:+1:

**Comment by 3rd-Eden** [comment]
@Swaagie Any thoughts on this? I'm neutral about it.

**Comment by roman01la** [comment]
:+1: I have separate parts of the project within a single repo, this will solve the issue.

**Comment by 3rd-Eden** [comment]
I guess we could make this functionality opt-in by specifying the max depth that we can search for a `.git` folder.

**Comment by tuhoojabotti** [comment]
The only complication I can see, if I have multiple projects in a repository in subfolders that use pre-commit.

**Comment by 3rd-Eden** [comment]
@tuhoojabotti That's why I'm suggesting opt-in through a field in the `package.json`

**Comment by tuhoojabotti** [comment]
Yeah, I don't see an issue with that, but would it be possible to run the pre-commit hooks of all packages in sub folders? For example if I have a package.json file for both my back-end and front-end I would like to run both pre-commit hooks before allowing commit.

**Comment by metaman** [comment]
:+1:

**Comment by katmai7** [comment]
+1

**Comment by trevordmiller** [comment]
+1

**Comment by kosmotaur** [comment]
Happy to give it a shot and create a PR. Could I get a hint please which part of the code to start with?

**Comment by 3rd-Eden** [comment]
@kosmotaur Probably the installation script of this module: https://github.com/observing/pre-commit/blob/master/install.js#L23-L28 as it checks if the parent folder has `.git`

**Comment by shaharmor** [comment]
Btw, when using git submodules there will be a file named `.git` in the same folder that will have a relative path to where the submodule real `.git` folder is.

Maybe this should be checked as well

**Comment by GuilhermeMedeiros** [comment]
:+1:

**Comment by fabricemathis** [comment]
Hello, we suffer from the same issue here, and I would like to know if there is some kind of improvement to come (like the opt-in suggested by @3rd-Eden) ?

**Comment by jacobangel** [comment]
This would be a great feature, especially for use with mono-repo projects like [Lerna](https://github.com/kittens/lerna). Is there any plans to merge #55, or ways we can pitch in?

**Comment by fabm22** [comment]
+1 Has this been addressed somehow ?

**Comment by JakeCooper** [comment]
+1!

**Comment by ai** [comment]
We need this fix in my project with [lint-staged](https://github.com/okonet/lint-staged/).

I think it is very important for big projects with monorepo, when you have `client/` and `server/` subdirs.

@3rd-Eden how I can help with this issue?

**Comment by TheWolfNL** [comment]
This [project](https://github.com/typicode/husky) sets up hooks which will work when .git is in a parent directory

**Comment by Alex-Sokolov** [comment]
👍

**Comment by nico-onmap** [comment]
I have a .git that's in a parent directory and it seems it's not detected by pre-commit which siletely fails when I installed it.

I tried creating a fake `.git/hooks` directory in the child directory where I run the npm installation and it works then.

Reading the comments I thought this was fixed?
